src/js/components/DataFilters/DataFilters.js

```diff
@@ -23,43 +30,61 @@ const layerProps = {
   position: 'right',
 };
 
-export const DataFilters = ({ drop, children, heading, layer, ...rest }) => {
+const defaultTouched = {};
+export const DataFilters = ({
+  drop,
+  children,
+  showClearFilters = true,
+  heading,
+  layer,
+  updateOn,
```

Comment:
We already have an `updateOn` prop on Data that seems to only be applied to DataFilters. When I tested filters that were outside of `DataFilters` they did not seem to be affected by Data's `updateOn` prop. This is a little confusing if we have both `updateOn` on Data and on DataFilters and they are trying both to control DataFilters. I think I would want to chat through the `updateOn` prop a bit more and how we envision it being applied to different fields. Related PR that would also be good to discuss within this context: https://github.com/grommet/grommet/pull/7029

Reply:
Good question. Summarizing our offline discussion here. Proposing that we remove `updateOn` at the global level since it doesn't provide much utility as soon as the caller enters into a compositional mode. It's also a bit cleaner to keep the property "close" to what it's effecting. In the case of `updateOn` it only affects DataFilters.

Updating the PR to reflect this.

Reply:
Pushed changes in this commit: https://github.com/grommet/grommet/pull/7025/commits/e323f3609045985bf6ed7242e2f74a809d894f64